pgmpy/factors/CPD.py

```diff
@@ -117,6 +117,21 @@ def __init__(self, variable, variable_card, values,
             raise TypeError("Values must be a 2D list/array")
 
         super(TabularCPD, self).__init__(variables, cardinality, values.flatten('C'))
+        
+    def __repr__(self):
+        if not self.evidence:
+            str_1 = ""
+            str_1 += '<TabularCPD representing P(' + str(self.variable) + ':' + str(self.variable_card) + ')>'
```

Comment:
Please don't use ```+``` operator with string objects. Its a costly operation as it create new objects. Instead you can use something like this

```python
str_1 = '<TabularCPD representing P({variable} : {cardinality})>'.format(variable=self.variable, cardinality=self.variable_card)
```